src/Workspaces/Remote/Core/Services/RemoteWorkspace.cs

```diff
@@ -25,6 +25,12 @@ public RemoteWorkspace()
             Options = Options.WithChangedOption(CacheOptions.RecoverableTreeLengthThreshold, 0);
         }
 
+        // constructor for testing
```

Comment:
Should the other constructor call this one or something? Why is the stuff in the other constructor not applicable in tests?